src/utils/index.ts

```diff
@@ -8,3 +11,57 @@ export const patchPnpmEnv = (inputs: Inputs): NodeJS.ProcessEnv => ({
   ...process.env,
   PATH: getBinDest(inputs) + path.delimiter + process.env.PATH,
 })
+
+/**
+ * Reads and parses a potential .npmrc file to extract registry configuration
+ * @param workspaceRoot - The root directory of the workspace
+ * @returns The registry URL if found, otherwise undefined
+ */
+export async function readNpmrcRegistry(workspaceRoot: string): Promise<string | undefined> {
+  try {
+    const npmrcPath = path.join(workspaceRoot, '.npmrc');
+
+    debug(`Attempting to read .npmrc file from: ${npmrcPath}`);
+    const content = await readFile(npmrcPath, 'utf8');
+
+    for (const line of content.split('\n')) {
```

Comment:
I'd prefer no `for` loop, no `.forEach` either. Just use `content.split('\n').map(line => line.trim()).filter(...).filter(...).find(...)`. It's not as efficient, yes, but in the context of `.npmrc`, the trade-off is worth it for readability.